pkg/iris/namespaces/services/events/eventProducer_test.go

```diff
@@ -4,18 +4,16 @@ import (
 	"crypto/sha256"
 	"encoding/hex"
 	"fmt"
-	"github.com/golang/mock/gomock"
 	"github.com/leanix/leanix-k8s-connector/pkg/iris/common/models"
 	common "github.com/leanix/leanix-k8s-connector/pkg/iris/common/services"
 	namespaceModels "github.com/leanix/leanix-k8s-connector/pkg/iris/namespaces/models"
+	"github.com/leanix/leanix-k8s-connector/pkg/mocks"
 	"github.com/stretchr/testify/assert"
 	"testing"
 )
 
 func Test_eventProducer_filter_created(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-	mockApi := NewMockAPI(ctrl)
+	mockApi := new(mocks.API)
```

Comment:
also can not we define this mock once and use it in every function?

Reply:
Tried but I can only access mock.Api

Reply:
@AlperCezibarakLeanix not sure, if we can do that in golang. @DennisBerthold-leanix please correct me if I'm wrong.

Reply:
`mockApi := mocks.NewIrisApi(t)`
That's what I meant. There is always a constructor which takes the Testing t to not panic when a test fails.
Now that we are using Testing t we can't describe the mockApi globally in the test because we need the Testing t which is being passed to every test itself. I commit the changes.